src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMigrationService.kt

```diff
@@ -51,39 +42,46 @@ class PrisonPersonMigrationService(
   completeCheckDelaySeconds = completeCheckDelaySeconds,
   completeCheckCount = completeCheckCount,
 ) {
+  abstract suspend fun migrateEntity(offenderNo: String): List<Long>
+
   override suspend fun getIds(
     migrationFilter: PrisonPersonMigrationFilter,
     pageSize: Long,
     pageNumber: Long,
-  ): PageImpl<PrisonerId> =
+  ): PageImpl<PrisonPersonMigrationRequest> =
     if (migrationFilter.prisonerNumber.isNullOrEmpty()) {
       nomisService.getPrisonerIds(
         pageNumber = pageNumber,
         pageSize = pageSize,
-      )
+      ).let {
+        PageImpl<PrisonPersonMigrationRequest>(
+          it.content.map { PrisonPersonMigrationRequest(it.offenderNo, migrationFilter.migrationType) },
+          it.pageable,
+          it.totalElements,
+        )
+      }
```

Comment:
We now decorate PrisonerId with the migration type which is held in each migrate entity message